go/coding_standard.md

```diff
@@ -31,18 +31,32 @@ Please add items here, or modify them as appropriate. We have git versioning, do
  * Many comments are not sentences, these should begin with a lower case letter and end without a period.
  * The first letter of sentences in comments are capitalized and ends with a period.
  
-## Git
-
- * Production repos never push to `master` or `develop` branches except the push-master.
- * `develop` should always pass tests.
- * Always start work with a "WIP: \*" (work-in-progress) pull-request, on a branch ideally named "feature/\*" or "hotfix/\*". You're encouraged to fork the repo and add it as a remote in .git/config (so you can push to your fork, but still use the same import paths to the original project).
- * When complete, pull requests should be opened to the `develop` branch.
- * Do not merge PRs that aren't final -- e.g. no code dead code or code that won't be used should be committed.
- * Squash and merge to keep the history clean.
- * Merging from `develop` to `master` happens during a release.
- * All branches should be in lowercase, underscores are okay in branch names
- * All pull requests should include any relevant additions to the `CHANGELOG.md`, If PR is to the develop branch changes
- can be included under an `unreleased changes` header at the top of the document.
+## Github
+
+Developement should following the regimented strucutre for branches and pull requests outlined herein. 
+For further inspiration checkout https://nvie.com/posts/a-successful-git-branching-model/.
+
+ * Production branches
+   * Production repos never push to `master` or `develop` branches except the push-master.
+   * `develop` should always pass tests.
+   * `develop` branch should have an on going open PR to `master`, high level comments can be made here
+   * Merging from `develop` to `master` happens during a release.
+ * Feature branches
+   * should following the naming convention `yourname/branch-description`
+   * all branches should be in lowercase, underscores/dashes are okay in branch names
+   * should have an open PR with the name "WIP: your feature description"
+   * All feature PRs should opened to merge to the `develop` branch, unless stacked on other PR leading to `develop`.
+    * Ideally, for faster merges, break your PR into several PRs that stack on top of each other and lead to `develop`.
+ * Reference branches
+   * Sometimes is is necessary to keep a branch with reference information even there there is no intention of merging 
+   * These branches should be named `yourname/ref/branch-description`
+ * Pull Requests (PRs)
```

Comment:
i'd add a note about codecov always going up or staying neutral. we need to stop merging PRs when codecov is going down.

Reply:
maybe works for UI code, but codecov is non-deterministic. For example, you can edit the README or docs in a repo, make a PR, and codecov will change. point being, this shouldn't be a requirement. docs aside, I've even seen a PR that was only tests being added that had codecov go down

Reply:
Another usability problem is that CodeCov doesn't update the report in the PR frequently enough, if at all, after the initial run. Maybe something we can confiugre or improve in another way.

Reply:
Codecov will only consider actual code if configured correctly. Changing the README will result in no change in codecov.

Reply:
@xla CodeCov should update after each run of the CI if configured correctly.

Reply:
@faboweb This hasn't been my experience, see #1237 as an example where the description from Codegov is not up-to-date with the latest commits.

Reply:
We could also make it part of the acceptance stage of a PR much as our normal CI. [Commit Status](https://docs.codecov.io/docs/commit-status) should make that possible.

Reply:
All this is the case in the voyager repo.
@greg-szabo I assume you set this up. Do you want to take a shot at this? We can pair, if you need assistance.

Reply:
@faboweb Could you link to the repo you mentioned?

Reply:
Sorry here: https://github.com/cosmos/voyager

Reply:
Inspired and shamelessly ripped off in tendermint/tendermint#1247.

Reply:
Cool I'm going to leave this point for somebody else to explain opening an issue: https://github.com/tendermint/coding/issues/55

Reply:
I had nothing to do with CodeDev. :) Accepting that it's non-deterministic freaks me out a bit. Maybe for now it should just be optional...
I have more faith in the upcoming Jenkins testing, but that takes some time...